webhook_server/tests/test_github_retry.py

```
@@ -0,0 +1,288 @@
+            side_effect=[
+                GithubException(status=500, data={"message": "Internal Server Error"}),
+                "success",
+            ]
+        )
+        result = await github_api_call(func, logger=mock_logger, log_prefix=LOG_PREFIX)
+        assert result == "success"
+        assert func.call_count == 2
+        mock_sleep.assert_called_once_with(2)
+
+    @patch("webhook_server.utils.github_retry.asyncio.sleep", new_callable=AsyncMock)
+    async def test_retry_on_github_500(self, mock_sleep, mock_logger):
```

Ruff (0.15.9)

[warning] Unused method argument: `mock_sleep`

(ARG002)